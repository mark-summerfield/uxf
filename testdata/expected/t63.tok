Import Str("http://localhost:5558/ttype-eg.uxf")
Import Str("ttype-test.uxi")
Import Str("complex")
Import Str("fraction")
Import Str("t63.uxt")
TClassBegin ttype=<point3d>
  Field Str("x") vtype=int
  Field Str("y") vtype=int
  Field Str("z") vtype=int
TClassEnd
TClassBegin ttype=<dob>
  Field Str("dob") vtype=date
TClassEnd
ListBegin
  TableBegin ttype=<img>
    TableBegin ttype=<rgb>
      Int(0)
      Int(0)
      Int(127)
    TableEnd
    Bytes( (:DECADE:) )
  TableEnd
  TableBegin ttype=<i>
    Str("is this oblique?")
  TableEnd
  MapBegin
    Int(10)
    TableBegin ttype=<point3d>
      Int(3)
      Int(4)
      Int(5)
      Int(5)
      Int(12)
      Int(13)
      Int(8)
      Int(15)
      Int(17)
      Int(7)
      Int(24)
      Int(25)
    TableEnd
    Int(20)
    TableBegin ttype=<cmyk>
      Real(0.9)
      Real(0.9)
      Real(0.5)
      Real(0.0)
    TableEnd
    Int(30)
    TableBegin ttype=<pair>
      Str("one")
      Int(2)
    TableEnd
  MapEnd
  TableBegin ttype=<Complex>
    Real(3.9)
    Real(-1.4)
    Real(-170.0)
    Real(2e-5)
  TableEnd
  TableBegin ttype=<Fraction>
    Int(22)
    Int(7)
  TableEnd
ListEnd
Eof
----------------------------------------
