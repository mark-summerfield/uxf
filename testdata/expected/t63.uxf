uxf 1.0
!http://www.qtrac.eu/ttype-eg.uxf
!ttype-test
!complex
!fraction
!t63.uxt
=dob dob:date
=point3d x:int y:int z:int
[
  (img (rgb 0 0 127) (:DECADE:)) 
  (i <is this oblique?>)
  {
    10
      (point3d
        3 4 5
        5 12 13
        8 15 17
        7 24 25
      )
    20 (cmyk 0.9 0.9 0.5 0.0)
    30 (pair <one> 2)
  }
  (Complex
    3.9 -1.4
    -170.0 2e-05
  ) 
  (Fraction 22 7)
]
