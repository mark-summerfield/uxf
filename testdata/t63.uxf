uxf 1.0
!http://www.qtrac.eu/ttype-eg.uxf
!ttype-test.uxi
!complex
!fraction
!t63.uxt
=point3d x:int y:int z:int
=dob dob:date
[
  (img (rgb 0 0 127) (:DECADE:))
  (i <is this oblique?>)
  {
      10 (point3d 3 4 5 5 12 13 8 15 17 7 24 25)
      20 (cmyk 0.9 0.9 0.5 0.0)
      30 (pair <one> 2)
  }
  (Complex 3.9 -1.4 -0.17e3 2e-5)
  (Fraction 22 7)
]
