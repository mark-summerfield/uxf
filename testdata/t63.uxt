uxf 1
=cmyk cyan:real magenta:real yellow:real black:real
=point2d x:int y:int
[]
