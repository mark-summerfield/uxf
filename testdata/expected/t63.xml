<?xml version="1.0" ?>
<uxf version="1">
  <imports>
    <import filename="http://localhost:5558/ttype-eg.uxf"/>
    <import filename="ttype-test.uxi"/>
    <import filename="complex"/>
    <import filename="fraction"/>
    <import filename="t63.uxt"/>
  </imports>
  <ttypes>
    <ttype name="B">
      <comment>bullet item</comment>
      <field name="content"/>
    </ttype>
    <ttype name="cmyk">
      <field name="cyan" vtype="real"/>
      <field name="magenta" vtype="real"/>
      <field name="yellow" vtype="real"/>
      <field name="black" vtype="real"/>
    </ttype>
    <ttype name="Complex">
      <field name="Real" vtype="real"/>
      <field name="Imag" vtype="real"/>
    </ttype>
    <ttype name="dob">
      <field name="dob" vtype="date"/>
    </ttype>
    <ttype name="Fraction">
      <field name="numerator" vtype="int"/>
      <field name="denominator" vtype="int"/>
    </ttype>
    <ttype name="h1">
      <comment>title</comment>
      <field name="content"/>
    </ttype>
    <ttype name="h2">
      <comment>subtitle</comment>
      <field name="content"/>
    </ttype>
    <ttype name="i">
      <comment>italic</comment>
      <field name="content"/>
    </ttype>
    <ttype name="img">
      <comment>image</comment>
      <field name="content"/>
      <field name="image" vtype="bytes"/>
    </ttype>
    <ttype name="IPv4">
      <field name="A" vtype="int"/>
      <field name="B" vtype="int"/>
      <field name="C" vtype="int"/>
      <field name="D" vtype="int"/>
    </ttype>
    <ttype name="m">
      <comment>monospace inline</comment>
      <field name="content"/>
    </ttype>
    <ttype name="nl">
      <comment>newline with no content</comment>
    </ttype>
    <ttype name="p">
      <comment>para</comment>
      <field name="content"/>
    </ttype>
    <ttype name="pair">
      <comment>a pair of any two values</comment>
      <field name="first"/>
      <field name="second"/>
    </ttype>
    <ttype name="point2d">
      <field name="x" vtype="int"/>
      <field name="y" vtype="int"/>
    </ttype>
    <ttype name="point3d">
      <field name="x" vtype="int"/>
      <field name="y" vtype="int"/>
      <field name="z" vtype="int"/>
    </ttype>
    <ttype name="pre">
      <comment>monospace block</comment>
      <field name="content"/>
    </ttype>
    <ttype name="rgb">
      <comment>0-255 per component</comment>
      <field name="red" vtype="int"/>
      <field name="green" vtype="int"/>
      <field name="blue" vtype="int"/>
    </ttype>
    <ttype name="rgba">
      <comment>0-255 per component</comment>
      <field name="red" vtype="int"/>
      <field name="green" vtype="int"/>
      <field name="blue" vtype="int"/>
      <field name="alpha" vtype="int"/>
    </ttype>
    <ttype name="Slide">
      <field name="title"/>
      <field name="body"/>
    </ttype>
    <ttype name="url">
      <field name="content"/>
      <field name="link"/>
    </ttype>
  </ttypes>
  <list>
    <table name="img">
      <row>
        <table name="rgb">
          <row>
            <int v="0"/>
            <int v="0"/>
            <int v="127"/>
          </row>
        </table>
        <bytes>DECADE</bytes>
      </row>
    </table>
    <table name="i">
      <row>
        <str>is this oblique?</str>
      </row>
    </table>
    <map>
      <key>
        <int v="10"/>
      </key>
      <value>
        <table name="point3d">
          <row>
            <int v="3"/>
            <int v="4"/>
            <int v="5"/>
          </row>
          <row>
            <int v="5"/>
            <int v="12"/>
            <int v="13"/>
          </row>
          <row>
            <int v="8"/>
            <int v="15"/>
            <int v="17"/>
          </row>
          <row>
            <int v="7"/>
            <int v="24"/>
            <int v="25"/>
          </row>
        </table>
      </value>
      <key>
        <int v="20"/>
      </key>
      <value>
        <table name="cmyk">
          <row>
            <real v="0.9"/>
            <real v="0.9"/>
            <real v="0.5"/>
            <real v="0.0"/>
          </row>
        </table>
      </value>
      <key>
        <int v="30"/>
      </key>
      <value>
        <table name="pair">
          <row>
            <str>one</str>
            <int v="2"/>
          </row>
        </table>
      </value>
    </map>
    <table name="Complex">
      <row>
        <real v="3.9"/>
        <real v="-1.4"/>
      </row>
      <row>
        <real v="-170.0"/>
        <real v="2e-05"/>
      </row>
    </table>
    <table name="Fraction">
      <row>
        <int v="22"/>
        <int v="7"/>
      </row>
    </table>
  </list>
</uxf>
